src/details/ArborX_DetailsNode.hpp

```diff
@@ -22,19 +22,27 @@ namespace Details
 {
 struct Node
 {
-  KOKKOS_INLINE_FUNCTION
-  Node() = default;
+  KOKKOS_FUNCTION
+  constexpr Node() = default;
 
-  KOKKOS_INLINE_FUNCTION
-  Node(const Kokkos::pair<Node *, Node *> &c, const Box &bb)
-      : children(c)
-      , bounding_box(bb)
+  KOKKOS_FUNCTION constexpr bool isLeaf() const noexcept
   {
+    return children.first == nullptr;
+  }
+
+  KOKKOS_FUNCTION std::size_t getLeafPermutationIndex() const noexcept
+  {
+    return reinterpret_cast<std::size_t>(children.second);
   }
 
   Kokkos::pair<Node *, Node *> children = {nullptr, nullptr};
   Box bounding_box;
 };
+
+KOKKOS_INLINE_FUNCTION Node makeLeafNode(std::size_t index, Box box) noexcept
```

Comment:
please rename `index` -> `permutation_index` for clarity

Reply:
Done.